narwhal/types/src/primary.rs

```diff
@@ -576,6 +602,170 @@ impl HeaderV1 {
     }
 }
 
+#[derive(Builder, Clone, Default, Deserialize, MallocSizeOf, Serialize)]
+#[builder(pattern = "owned", build_fn(skip))]
+pub struct HeaderV2 {
```

Comment:
I think it doesn't really matter in this case because HeaderV1 is just a HeaderV2 without SystemMessages?

Reply:
Yeah thats fair, I see now you returned an empty `Vec<SystemMessage>` for `HeaderV1`. I have been adding panics for unsupported fields in new NW versioned types so I just assumed it was the case here 😅  But an empty message is perfectly valid in this case

Reply:
Might be worthwhile to add a comment somewhere that the versions are backwards compatible in that way, in case there are some confusions in debugging with mixed header versions